### PR TITLE
Feature: Upload file to new project via onboarder

### PR DESCRIPTION
**Pull request by samanehsan**
Continuation of #2411 

From @kushG: 
><h3> Purpose </h3>
Currently, Onboarder allows a user to either create a new project or upload file(s) on an existing project. There's no way by which the user can create a new project & upload file(s) in that same project. This PR allows uploading file(s) on a newly created project.

>Resolves https://github.com/CenterForOpenScience/osf.io/issues/2146

><h3> Changes </h3>
Adds couple of more VM javascript functions to handle project creation inside OBUploaderViewModel. project_new_post view function updated to return node serialized data.

><h3> Side Effects </h3>
View function modification might have missed an edge case

### CODE REVIEW

**Comment by samanehsan** [comment]
Tagged as a work in progress because I need to make this work with Lauren's latest changes to adding a pop-up warning when you try to navigate away from uploading a file.

**Comment by samanehsan** [comment]
This is ready for review @sloria! :whale:

File: website/routes.py [hunk 1]
```diff
@@ -907,6 +907,7 @@ def make_url_map(app):
 
         # Create project, used by projectCreator.js
         Rule('/project/new/', 'post', project_views.node.project_new_post, json_renderer),
+        Rule('/project/new/delete/<pid>/', 'delete', project_views.node.project_delete_new_post, json_renderer),
```
Comment (sloria) [inline website/routes.py]: This route seems unnecessary. Can you use the existing route for deleting nodes? https://github.com/CenterForOpenScience/osf.io/blob/e1a2d38ed8c9a8c2cd4b6d6126f1c176e9af8bcd/website/routes.py#L1101-1110

File: website/static/js/onboarder.js [hunk 2]
```diff
@@ -440,6 +445,24 @@ function OBUploaderViewModel(params) {
         }
         if (!self.dropzone.getQueuedFiles().length) {
             self.changeMessage('Please select at least one file to upload.', 'text-danger');
+
+            // Delete node only if a new project is created
+            if(self.newProjectName() != null)
```
Comment (sloria) [inline website/static/js/onboarder.js]: For consistency and readability:

```js
if (self.newProjectName()) {
    //
}
```

File: website/static/js/onboarder.js [hunk 3]
```diff
@@ -440,6 +445,24 @@ function OBUploaderViewModel(params) {
...
+            {
+                var request = $.ajax({
+                url: DELETE_NODE_URL + '/' + selectedProject.id + '/',
+                type: 'DELETE'
+                });
+                request.done(function() {
```
Comment (sloria) [inline website/static/js/onboarder.js]: This callback seems unnecessary.

File: website/static/js/onboarder.js (hunk 2 above)
Comment (sloria) [inline website/static/js/onboarder.js]: Also, I'm not clear on this logic. Why do we delete the selected project here?

Comment (sloria) [inline website/static/js/onboarder.js]: Oh, I see, we delete the project if the user tries to start an upload to a new project without first selecting a file.

Is there any way we can prevent the creation of a new project before we get to this point?

Comment (sloria) [inline website/static/js/onboarder.js]: For example, the logic for submitting the form should be:

- Iff a file is selected and an existing node is selected: upload the file to the node
- Iff a files is selected and new node title (not from the dropdown) is entered: create the new node and upload the file to it

In all other cases, either an error message is shown or submission is disallowed.

**Comment by sloria** [comment]
Initial pass finished. :point_up:

**Comment by samanehsan** [comment]
Done making those changes @sloria! :snake: The failing tests are also failing on develop at the moment. 

Also, since the "Upload" button does not show up until you enter a project to upload to, is it fine if I do the same for the "Create" button?

**Comment by wisherp** [comment]
@samanehsan @sloria I am still unable to upload files through the onboarder on IE 10 (through UVA Hive) and IE 11 (through Virtualbox). Issue's both on production and staging

**Comment by sloria** [comment]
@wisherp That is expected and not caused by this PR. I believe there is another issue re: the uploader's incompatibility with IE

**Comment by lbanner** [comment]
This -- I think: https://github.com/CenterForOpenScience/osf.io/issues/1669

**Pull request merged**